kolibri/core/assets/src/views/FullScreenSidePanel/index.vue

```diff
@@ -171,24 +184,44 @@
         };
       },
     },
+    beforeMount() {
+      this.lastFocus = document.activeElement;
+    },
     /* this is the easiest way I could think to avoid having dual scroll bars */
     mounted() {
       const htmlTag = window.document.getElementsByTagName('html')[0];
       htmlTag.style['overflow-y'] = 'hidden';
       // Gets the height of the fixed header - adds 40 to account for padding
       this.fixedHeaderHeight = this.$refs.fixedHeader.clientHeight + 'px';
 
-      // Ensures user starts at top header with keyboard focus
-      this.$refs.fixedHeader.focus();
+      this.$nextTick(() => {
+        this.setFocusTrap();
+      });
     },
     beforeDestroy() {
       const htmlTag = window.document.getElementsByTagName('html')[0];
       htmlTag.style['overflow-y'] = 'auto';
     },
+    destroyed() {
+      window.setTimeout(() => this.lastFocus.focus());
+    },
     methods: {
       closePanel() {
         this.$emit('closePanel');
       },
+      setFocusTrap() {
+        if (this.$refs.sidePanel && !this.$refs.sidePanel.contains(document.activeElement)) {
+          this.focusSidePanel();
+        }
+        this.firstFocusableEl =
+          this.$el.querySelector('.search-box') || // if `EmbeddedSidePanel` is for search/filter btn
```

Comment:
I'd recommend moving focus logic specific to the embedded side panel to the `EmbeddedSidePanel` component as we chatted briefly earlier. These dependencies seem to be brittle and I can imagine that focus logic could stop working silently - for example, if we updated some classes in the `EmbeddedSidePanel`, we may easily not notice that some updates are needed in the `FullScreenSidePanel` as well.

I recall that you mentioned that we don't want to use focus trapping when `EmbeddedSidePanel` is used on its own, outside of `FullScreenSidePanel`. I'm not sure which way would be better feasible here off the top of my head, but can imagine two directions (their combination might be needed) to play with that could help us with moving the `FocusTrap` component and related to the `EmbeddedSidePanel`:

- Providing `setFocusTrap` as a public method from the `EmbeddedSidePanel` to allow other components to call it (you can search for JSDoc `@public` tag in our codebase to see some examples of how we use public methods)
- Adding `Boolean` `enableFocusTrap` property to the `EmbeddedSidePanel` while setting it to `true` only when used inside of the `FullScreenSidePanel`. Focus-related logic would then be only run in the `EmbeddedSidePanel` when `enableFocusTrap` would be `true` and skipped otherwise. It should be possible to propagate this information into the `FocusTrap` component since it provides `disabled` property.

Reply:
Going off what @MisRob said, a public method on `EmbeddedSidePanel` to focus it on the appropriate internal element makes sense to me. 

There's a pattern we can follow in plain HTML-- you can think of `<input>` as just another Vue component, so when you want to focus on it, you call that method `.focus()`. That allows the component to decide how to handle that appropriately.

The other case is when the element or Vue component wraps other elements or components, like the `FullScreenSidePanel`. In this instance, you want it to trap the focus, so implementing `FocusTrap` would either be inside this component or in whatever component defines the wrapped content, since that component has access to its children. In this case, I think the pattern is to use events. So `FullScreenSidePanel` could emit and event like `@shouldFocus` to trigger focusing on the appropriate element.

Although, the current `FocusTrap` implementation relies on access to the component children, which conflicts with the use case of a wrapper component. To make it work with this approach, it could also emit events when the first and last elements should be focused and avoid requiring the elements as props.

So as an example, it could look like this:
```html
<FullScreenSidePanel @shouldFocus="$refs.embedded.focus()">
  <EmbeddedSidePanel ref="embedded" />
</FullScreenSidePanel>
```
where `$emit("shouldFocus")` occurs in the `$nextTick` on `mount()` and on `<FocusTrap @focusFirstEl="$emit('shouldFocus')">` if it was modified to emit events.

Reply:
Thank you @bjester, I appreciate that with this solution, it sounds like we wouldn't need `enableFocusTrap` prop on the `EmbeddedSidePanel` and it'd still be `FullScreenSidePanel` that's responsible for controlling most of the focus logic which makes sense to me since we need to handle focus only when we use `FullScreenSidePanel`, and having it all in the `EmbeddedSidePanel` might not be relevant in cases we use it outside of the `FullScreenSidePanel` context.